src/test/java/com/salaboy/conferences/c4p/controller/C4PControllerIT.java

```diff
@@ -0,0 +1,99 @@
+import io.zeebe.spring.client.ZeebeClientLifecycle;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.reactive.WebFluxTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.web.reactive.server.WebTestClient;
+import org.springframework.web.reactive.function.BodyInserters;
+
+import static org.mockito.Answers.RETURNS_DEEP_STUBS;
+
+@ExtendWith(SpringExtension.class)
+@WebFluxTest(controllers = C4PController.class)
+public class C4PControllerIT {
+
+    @Autowired
+    private WebTestClient webTestClient;
+
+    @MockBean
+    private ProposalRepository proposalRepository;
+
+    @MockBean(answer = RETURNS_DEEP_STUBS)
+    private ZeebeClientLifecycle zeebeClientLifecycle;
+
+    @MockBean(answer = RETURNS_DEEP_STUBS)
+    private ZeebeClientBuilder zeebeClientBuilder;
+
+    private WebTestClient.ResponseSpec createProposalRequest() {
+
+        var requestProposal =
+                new Proposal("Title", "Description", "Author", "email@email.com");
+
+        Mockito.when(proposalRepository.save(requestProposal)).thenReturn(requestProposal);
+
+        return webTestClient.post()
+                .uri("/")
+                .contentType(MediaType.APPLICATION_JSON)
+                .body(BodyInserters.fromValue(requestProposal))
+                .exchange()
+                .expectStatus().isOk();
+    }
+
+    private WebTestClient.ResponseSpec getAllProposals() {
+        return webTestClient.get()
+                .uri("/")
+                .exchange();
+    }
+
+    @Test
+    public void newProposal_ShouldBeCreateAProposal() {
+
+        // action
+        var responseSpec = createProposalRequest();
+
+        // assert
+        responseSpec.expectBody(Proposal.class);
+    }
+
+    @Test
+    public void deleteProposal_ShouldDeleteProposalById() {
+
+        // arrange
+        var responseSpec = createProposalRequest();
+        var proposal = responseSpec.expectBody(Proposal.class).returnResult().getResponseBody();
+
+        // action, assert
+        webTestClient.delete()
+                .uri("/" + proposal.getId())
+                .exchange()
+                .expectStatus().isOk();
+
+        getAllProposals().expectBodyList(Proposal.class).hasSize(0);
+    }
+
+    @Test
+    public void deleteProposals_ShouldDeleteAllProposals() {
+
+        // arrange
+        createProposalRequest();
+        createProposalRequest();
+
+        // action, assert
+        webTestClient.delete()
+                .uri("/")
+                .exchange()
+                .expectStatus().isOk();
+
+        getAllProposals().expectBodyList(Proposal.class).hasSize(0);
+    }
```

Comment:
@mcruzdev we should have a test here for the decision endpoint.. that is the most tricky one.. can you please add that one?